python/lsst/pipe/base/graph/_implDetails.py

```diff
@@ -40,58 +39,61 @@
 _U = TypeVar("_U", TaskDef, QuantumNode)
 
 
-@dataclass
-class _DatasetTrackerElement(Generic[_U]):
-    inputs: Set[_U] = field(default_factory=set)
-    output: Optional[_U] = None
-
-
 class _DatasetTracker(Generic[_T, _U]):
```

Comment:
The class and the methods are missing docstrings.  I guess because this is internal that might be okay, but I think it'll make it clearer when you look at this code in 6 months.

Reply:
As of now I still remember this from more than 6 months ago, but you are right more documentation is better, especially if it is not me that looks at this next time.